jsapp/js/components/special/koboAccessibleSelect.spec.tsx

```diff
@@ -0,0 +1,136 @@
+import {fireEvent, render, screen} from '@testing-library/react';
+import {describe, it, expect, jest} from '@jest/globals';
+import userEvent from '@testing-library/user-event';
+import type {KoboSelectOption} from './koboAccessibleSelect';
+import KoboSelect3 from './koboAccessibleSelect';
+import {useState} from 'react';
+
+const options: KoboSelectOption[] = [
+  {value: '1', label: 'Apple'},
+  {value: '2', label: 'Banana'},
+  {value: '3', label: 'Avocado'},
+];
+
+// A wrapper is needed for the component to retain value changes
```

Comment:
📓 Good point — many of our Kobo components are like this — 'controlled'-only. It'd be possible to make them more flexible.

- Cool article: [Creating React Components that can be Controlled and Uncontrolled](https://maxschmitt.me/posts/react-components-controlled-uncontrolled)

'Course, then there would be more to test :sweat_smile:

Reply:
It's a common practice I found for single component testing, since most of the components will depend on external context anyways. So I really don't think it's the case of making them 'uncontrolled' just for the sake of tests. We only need to be careful to not create biased tests that could influence the component behavior. (but yeah, cool article! 😄)